### ISSUE TITLE
Bug: [scope-manager] Static Blocks do not introduce a scope

### ISSUE THREAD

**Issue by DMartens** [typescript-eslint/typescript-eslint#5485]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

scope-manager

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&showAST=scope&code=MYGwhgzhAEDC5WgbwFDWhALmTBLYy0AviiUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Class {
    static {}
}
```

### Expected Result

I expected the static block to have its own scope but no scope is introduced.

### Actual Result

There is no scope for the static block, but eslint-scope adds a scope (relevant [PR](https://github.com/eslint/eslint-scope/pull/80/files)).

### Additional Info

A class for static blocks [exists](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/scope-manager/src/scope/ClassStaticBlockScope.ts) but the [ClassVisitor](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/scope-manager/src/referencer/ClassVisitor.ts) does not reference it.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`     | `5.3.3` |
